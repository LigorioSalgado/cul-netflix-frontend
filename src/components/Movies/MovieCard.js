import React, { Component } from 'react'
import Rating from 'react-rating';

export default class MovieCard extends Component {
  render() {


    return (
        <div className="card">
            <div className="card-image waves-effect waves-block waves-light">
                <a href={`/movie/${this.props.id}`}>
                    <img className="activator" src={this.props.image} alt="movie photo"/>
                </a>
            </div>
            <div className="card-content">
                <span className="card-title activator grey-text text-darken-4">
                    {this.props.title}
                </span>
            
                <Rating
                    initialRating={this.props.rating}
                    readonly
                />
            </div>
      </div>
    )
  }
}
