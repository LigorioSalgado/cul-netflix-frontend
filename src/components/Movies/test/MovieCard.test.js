import React from 'react';
import {shallow} from "enzyme";

import MovieCard from '../MovieCard';



describe("<MovieCard/>", () => {

    it("Debe ejecutar render correctamente",() => {
        const component =  shallow(<MovieCard/>)
        expect(component).toMatchSnapshot();
    })

    it("Debe pintar props",() => {
        const movie = {
            id:"pruebaid",
            image:"pruebaimagen.jpg",
            rating:4.5,
            title:"prueba pelicula"
        }

         const component =  shallow(<MovieCard id={movie.id} 
                image={movie.image}
                rating={movie.rating}
                title={movie.title}
             />)
        
        expect(component.find(".card-title").text()).toBe(movie.title);
   })

})