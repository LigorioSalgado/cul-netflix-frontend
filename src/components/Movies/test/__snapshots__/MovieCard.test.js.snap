// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard/> Debe ejecutar render correctamente 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-image waves-effect waves-block waves-light"
        >
          <a
            href="/movie/undefined"
          >
            <img
              alt="movie photo"
              className="activator"
            />
          </a>
        </div>,
        <div
          className="card-content"
        >
          <span
            className="card-title activator grey-text text-darken-4"
          />
          <RatingAPILayer
            direction="ltr"
            emptySymbol={
              Object {
                "backgroundColor": "#ccc",
                "border": "5px double white",
                "borderRadius": "50%",
                "display": "inline-block",
                "height": 30,
                "width": 30,
              }
            }
            fractions={1}
            fullSymbol={
              Object {
                "backgroundColor": "black",
                "border": "5px double white",
                "borderRadius": "50%",
                "display": "inline-block",
                "height": 30,
                "width": 30,
              }
            }
            onChange={[Function]}
            onClick={[Function]}
            onHover={[Function]}
            placeholderSymbol={
              Object {
                "backgroundColor": "red",
                "border": "5px double white",
                "borderRadius": "50%",
                "display": "inline-block",
                "height": 30,
                "width": 30,
              }
            }
            quiet={false}
            readonly={true}
            start={0}
            step={1}
            stop={5}
          />
        </div>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="/movie/undefined"
          >
            <img
              alt="movie photo"
              className="activator"
            />
          </a>,
          "className": "card-image waves-effect waves-block waves-light",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="movie photo"
              className="activator"
            />,
            "href": "/movie/undefined",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "movie photo",
              "className": "activator",
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="card-title activator grey-text text-darken-4"
            />,
            <RatingAPILayer
              direction="ltr"
              emptySymbol={
                Object {
                  "backgroundColor": "#ccc",
                  "border": "5px double white",
                  "borderRadius": "50%",
                  "display": "inline-block",
                  "height": 30,
                  "width": 30,
                }
              }
              fractions={1}
              fullSymbol={
                Object {
                  "backgroundColor": "black",
                  "border": "5px double white",
                  "borderRadius": "50%",
                  "display": "inline-block",
                  "height": 30,
                  "width": 30,
                }
              }
              onChange={[Function]}
              onClick={[Function]}
              onHover={[Function]}
              placeholderSymbol={
                Object {
                  "backgroundColor": "red",
                  "border": "5px double white",
                  "borderRadius": "50%",
                  "display": "inline-block",
                  "height": 30,
                  "width": 30,
                }
              }
              quiet={false}
              readonly={true}
              start={0}
              step={1}
              stop={5}
            />,
          ],
          "className": "card-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "card-title activator grey-text text-darken-4",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "ltr",
              "emptySymbol": Object {
                "backgroundColor": "#ccc",
                "border": "5px double white",
                "borderRadius": "50%",
                "display": "inline-block",
                "height": 30,
                "width": 30,
              },
              "fractions": 1,
              "fullSymbol": Object {
                "backgroundColor": "black",
                "border": "5px double white",
                "borderRadius": "50%",
                "display": "inline-block",
                "height": 30,
                "width": 30,
              },
              "initialRating": undefined,
              "onChange": [Function],
              "onClick": [Function],
              "onHover": [Function],
              "placeholderSymbol": Object {
                "backgroundColor": "red",
                "border": "5px double white",
                "borderRadius": "50%",
                "display": "inline-block",
                "height": 30,
                "width": 30,
              },
              "quiet": false,
              "readonly": true,
              "start": 0,
              "step": 1,
              "stop": 5,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-image waves-effect waves-block waves-light"
          >
            <a
              href="/movie/undefined"
            >
              <img
                alt="movie photo"
                className="activator"
              />
            </a>
          </div>,
          <div
            className="card-content"
          >
            <span
              className="card-title activator grey-text text-darken-4"
            />
            <RatingAPILayer
              direction="ltr"
              emptySymbol={
                Object {
                  "backgroundColor": "#ccc",
                  "border": "5px double white",
                  "borderRadius": "50%",
                  "display": "inline-block",
                  "height": 30,
                  "width": 30,
                }
              }
              fractions={1}
              fullSymbol={
                Object {
                  "backgroundColor": "black",
                  "border": "5px double white",
                  "borderRadius": "50%",
                  "display": "inline-block",
                  "height": 30,
                  "width": 30,
                }
              }
              onChange={[Function]}
              onClick={[Function]}
              onHover={[Function]}
              placeholderSymbol={
                Object {
                  "backgroundColor": "red",
                  "border": "5px double white",
                  "borderRadius": "50%",
                  "display": "inline-block",
                  "height": 30,
                  "width": 30,
                }
              }
              quiet={false}
              readonly={true}
              start={0}
              step={1}
              stop={5}
            />
          </div>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="/movie/undefined"
            >
              <img
                alt="movie photo"
                className="activator"
              />
            </a>,
            "className": "card-image waves-effect waves-block waves-light",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="movie photo"
                className="activator"
              />,
              "href": "/movie/undefined",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "movie photo",
                "className": "activator",
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="card-title activator grey-text text-darken-4"
              />,
              <RatingAPILayer
                direction="ltr"
                emptySymbol={
                  Object {
                    "backgroundColor": "#ccc",
                    "border": "5px double white",
                    "borderRadius": "50%",
                    "display": "inline-block",
                    "height": 30,
                    "width": 30,
                  }
                }
                fractions={1}
                fullSymbol={
                  Object {
                    "backgroundColor": "black",
                    "border": "5px double white",
                    "borderRadius": "50%",
                    "display": "inline-block",
                    "height": 30,
                    "width": 30,
                  }
                }
                onChange={[Function]}
                onClick={[Function]}
                onHover={[Function]}
                placeholderSymbol={
                  Object {
                    "backgroundColor": "red",
                    "border": "5px double white",
                    "borderRadius": "50%",
                    "display": "inline-block",
                    "height": 30,
                    "width": 30,
                  }
                }
                quiet={false}
                readonly={true}
                start={0}
                step={1}
                stop={5}
              />,
            ],
            "className": "card-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "card-title activator grey-text text-darken-4",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "ltr",
                "emptySymbol": Object {
                  "backgroundColor": "#ccc",
                  "border": "5px double white",
                  "borderRadius": "50%",
                  "display": "inline-block",
                  "height": 30,
                  "width": 30,
                },
                "fractions": 1,
                "fullSymbol": Object {
                  "backgroundColor": "black",
                  "border": "5px double white",
                  "borderRadius": "50%",
                  "display": "inline-block",
                  "height": 30,
                  "width": 30,
                },
                "initialRating": undefined,
                "onChange": [Function],
                "onClick": [Function],
                "onHover": [Function],
                "placeholderSymbol": Object {
                  "backgroundColor": "red",
                  "border": "5px double white",
                  "borderRadius": "50%",
                  "display": "inline-block",
                  "height": 30,
                  "width": 30,
                },
                "quiet": false,
                "readonly": true,
                "start": 0,
                "step": 1,
                "stop": 5,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
